nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/ProcessorNode.java

```diff
@@ -290,4 +290,11 @@ public abstract void runOnce(ScheduledExecutorService scheduler, long administra
      * @return the desired state for this Processor
      */
     public abstract ScheduledState getDesiredState();
+
+    /**
+     * This method will be called once the processor's configuration has been restored (on startup, reload, e.g.)
+     *
+     * @param context The ProcessContext associated with the Processor configuration
+     */
+    public abstract void onConfigurationRestored(ProcessContext context);
```

Comment:
Curious, why does `ProcessorNode` get an `onConfigurationRestored` method, but not `ControllerServiceNode` or `ReportingTaskNode`?

Reply:
They will eventually need one, but I think not right now because of [NIFI-5547](https://issues.apache.org/jira/browse/NIFI-5547)